### PR TITLE
Dv by child entity

### PR DESCRIPTION
**Pull request by chenshoo**

### CODE REVIEW

File: packages/polaris-common/src/context/data-version-context.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export interface DataVersionContext {
+    mapping?: Map<any, any>;
```
Comment (osher-sade) [inline packages/polaris-common/src/context/data-version-context.ts]: change the key to type `string`

File: packages/polaris-middlewares/src/data-version/data-version-middleware.ts [hunk 2]
```diff
@@ -82,4 +113,41 @@ export class DataVersionMiddleware {
...
+                    key = selection.name.value;
+                }
+                const res = this.loadDVRelations(newRoot, key, newInfo, rootInfo);
+                this.pushDVMapping(relations, key, res);
+            }
+        }
+        return relations.size > 0 ? relations : undefined;
+    }
+
+    pushDVMapping(map: Map<any, any>, key: any, value: any) {
```
Comment (osher-sade) [inline packages/polaris-middlewares/src/data-version/data-version-middleware.ts]: add `private` modifier

File: packages/polaris-typeorm/src/handlers/data-version-handler.ts [hunk 3]
```diff
@@ -66,3 +69,94 @@ export class DataVersionHandler {
...
+    entityMetadata: EntityMetadata,
+) {
+    const childDVMapping = getPropertyMap(children, relation);
+    if (childDVMapping) {
+        const relationMetadata = relation.inverseEntityMetadata;
+        const alias: string = relationMetadata.tableName;
+        const notInJoins = names.filter((x) => x === alias).length === 0;
+        if (isDescendentOfCommonModel(relationMetadata) && notInJoins) {
+            qb = qb.leftJoinAndSelect(
+                entityMetadata.tableName + '.' + relation.propertyName,
```
Comment (osher-sade) [inline packages/polaris-typeorm/src/handlers/data-version-handler.ts]: use template string

File: packages/polaris-typeorm/src/handlers/data-version-handler.ts [hunk 4]
```diff
@@ -66,3 +69,94 @@ export class DataVersionHandler {
...
+) => {
+    if (context.requestHeaders.dataVersion && context.requestHeaders.dataVersion > 0) {
+        qb = qb.distinct();
+        const entityMetadata = connection.getMetadata(entity);
+        let names = [entity];
+        if (context.dataVersionContext?.mapping) {
+            qb = loadRelations(qb, entityMetadata, names, context.dataVersionContext!.mapping!);
+        }
+        const dataVersion = context.requestHeaders.dataVersion;
+        qb.where(entity + '.dataVersion > :dataVersion', { dataVersion });
```
Comment (osher-sade) [inline packages/polaris-typeorm/src/handlers/data-version-handler.ts]: use template string

File: packages/polaris-typeorm/src/handlers/data-version-handler.ts [hunk 5]
```diff
@@ -66,3 +69,94 @@ export class DataVersionHandler {
...
+        names = names.slice(1);
+        for (const name of names) {
+            qb = qb.orWhere(name + '.' + 'dataVersion > :dataVersion');
```
Comment (osher-sade) [inline packages/polaris-typeorm/src/handlers/data-version-handler.ts]: use template string

**Review by osher-sade — APPROVED** [review]

File: packages/polaris-test/tests/disable-data-version-filter.test.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { createServers } from '../test-utils/tests-servers-util';
+import { graphQLRequest } from '../test-utils/graphql-client';
+import * as authors from './jsonRequestsAndHeaders/authors.json';
+import { createAuthorAndBook, createPen } from './data-version-filter.test';
+
+describe('enable data version filter', () => {
+    // TODO: figure out how to make the config to not be burnt to the scope,
```
Comment (itaykl) [inline packages/polaris-test/tests/disable-data-version-filter.test.ts:7]: is this done? or maybe you forgot to delete this comment?

File: packages/polaris-typeorm/test/integration-tests/utils/test-properties.ts [hunk 7]
```diff
@@ -22,7 +22,7 @@ export const applicationLogProperties: ApplicationProperties = {
 };
 
 export const loggerConfig: LoggerConfiguration = {
-    loggerLevel: LoggerLevel.INFO,
+    loggerLevel: LoggerLevel.WARN,
```
Comment (itaykl) [inline packages/polaris-typeorm/test/integration-tests/utils/test-properties.ts:25]: we want that warn will be our default?

**Review by itaykl — APPROVED** [review]

**Pull request merged**